app/src/main/java/com/leoart/koreanphrasebook/KoreanPhrasebookApp.java

```diff
@@ -19,5 +40,27 @@ public void onCreate() {
                 .deleteRealmIfMigrationNeeded()
                 .build();
         Realm.setDefaultConfiguration(config);
+
+        analyticsComponent = DaggerAnalyticsComponent
+                .builder()
+                .appModule(new AppModule(this))
+                .analyticsModule(new AnalyticsModule())
+                .build();
+        analyticsComponent.inject(this);
+    }
+
+    public AnalyticsComponent getAnalyticsComponent() {
+        return analyticsComponent;
     }
+
+    @Override
+    public AndroidInjector<Activity> activityInjector() {
+        return dispatchingAndroidInjector;
+    }
+
+    @Override
+    public AndroidInjector<Fragment> supportFragmentInjector() {
```

Comment:
AndroidInjectors should honor android hierarchy. For example, injector for activity should be in application, but injector for fragment should be in root fragment activity